src/index.js

```diff
@@ -67,7 +67,8 @@ export default function withSideEffect(
         return recordedState;
       }
 
-      UNSAFE_componentWillMount() {
+      constructor(props, context) {
```

Comment:
Are you sure this should be in the constructor instead of `componentDidMount`? Ideally, constructors should not have side-effects